tests/tools/onert_train/src/args.cc

```diff
@@ -184,6 +198,17 @@ void Args::Initialize(void)
     ("nnpackage", po::value<std::string>()->notifier(process_nnpackage), "NN Package file(directory) name")
     ("modelfile", po::value<std::string>()->notifier(process_modelfile), "NN Model filename")
     ("path", po::value<std::string>()->notifier(process_path), "NN Package or NN Modelfile path")
+    ("data_length", po::value<int>()->default_value(-1)->notifier([&](const auto &v) { _data_length = v; }), "Data length number")
+    ("input_datafile", po::value<std::string>()->notifier(process_input_datafile),
+         "NN Model Input data file\n"
+         "The datafile must have data for each input number.\n"
+         "If there are 3 inputs, the data of input0 must exist as much as data_length, "
+         "and the data for input1 and input2 must be held sequentially as data_length.\n"
+    )
+    ("expected_datafile", po::value<std::string>()->notifier(process_expected_datafile),
+         "NN Model Expected data file\n"
+         "(Same data policy with input_datafile)\n"
+    )
```

Comment:
@jyoungyun I don't understand the input and expected data file format. I guess it may need more information. For your information, `onert_run` used hdf5 format for input and output file. `--dump` and `--load` are the options.

Reply:
`onert_run` takes both file formats hdf5 and raw format. And this code is related to raw format data files. Besides I haven't considered hdf5 datafile format yet. The `--dump` option in `onert_run` is for output data. However, this code is for loading input and expected data file, so I can't use `--dump` here.

I can change the argument name to `--load_input:raw` and `--load_expected:raw`.
Do you want to change the argument name like this?

Reply:
@glistening Yes, I forgot to mention raw format. ( I wrote the support for both hdf5 and raw. ) Why don't you use --load:raw? I've found https://github.com/Samsung/ONE/pull/9016.

Reply:
@jyoungyun I am not sure the meaning of `raw`. It does not alter the format, it just read and write as-it-is. But where will you get the expected format?

Reply:
> But where will you get the expected raw values?

I got the raw value from `onert_run` with `--dump:raw` argument and small changes to dump raw input data.
The meaning of `raw` is the same with `onert_run`.

Reply:
Ah... I thought the explanation

>   "The datafile must have data for each input number.\n"
>   "If there are 3 inputs, the data of input0 must exist as much as data_length, "
>   "and the data for input1 and input2 must be held sequentially as data_length.\n"

is about file format.

But it seems about multiple input files.

@jyoungyun What does `data_length` mean? Input count?

If so, how can I pass the multiple input files using `load_input:raw`?
I don't understand how to use this parameter.

Could you please show me an example usage?

Reply:
> @jyoungyun What does data_length mean? Input count?

`data_length` means how many input data in `load_input:raw` file.

You can use this argument like below:

```
$ onert_train --modelfile a.circle --data_length 100 --load_input:raw a.in.bin --load_expected:raw a.ex.bin
```

Reply:
> @jyoungyun In above case, the input data file names should be:
> a.in.bin1, a.in.bin2, ..., a.in.bin100 ?
> 
> Am I right?

No, in the above case, `a.in.bin` and `a.ex.bin` contain 100 counts data. So we need `data_length`.
The one datafile contains both data as much as data_length and input counts.

Reply:
Ah, is it for batch_size? Then, how multiple inputs are handled?

Reply:
It is not a batch_size. The data_length will be greater than batch_size.
And I described how to handle multiple inputs in this PR.

```
         "The datafile must have data for each input number.\n"
         "If there are 3 inputs, the data of input0 must exist as much as data_length, "
         "and the data for input1 and input2 must be held sequentially as data_length.\n"
```

If you have 3 inputs in a model and the data size is 100, your data file will be configured like this:

```
in0_0 in0_1 in0_2 ... in0_(data_length-1) in1_0 in1_1 in1_2 ... in1_(data_length-1) in2_0 in2_1 in2_2 ... in2_(data_length-1)
```

Reply:
I could not understand how to use from your PR's argument description. Now, with more explanation especially with the example above, I understood. 

1. But argument description lacks of the information of `_#` for each input.

2. Do we need to specify in0_0 in0_1 ... in0_{data_length_1) ?
We may choose to try in0_0 to in0_{data_length_1} automatically if `data_length` is passed by user.

I would like to get other's opinion.

Reply:
> No, in the above case, `a.in.bin` and `a.ex.bin` contain 100 counts data. So we need `data_length`. The one datafile contains both data as much as data_length and input counts.

Then, as I understand, `data_length` is about dataset. `data_length` means the count of training examples.

Reply:
> you mean that the user needs to prepare those files, not to type those in command line.

Yes, the data file contains the data with the above format.

> Then, as I understand, data_length is about dataset. data_length means the count of training examples.

Yes, it is what I want to say. :)

Reply:
@glistening I made a draft PR for generating data file. This readme file will help you understand the format of datafile.
https://github.com/Samsung/ONE/pull/10901/files#diff-bbfbb7c2736d349f694e20894520282f5478b13cdf499e02f20b2e890e6d37b8

Reply:
Thank you for your draft. It is the same to what I've understood before your draft PR.

At first, I was confused it is about file format. But, as I understand, it does not introduce new format, it uses the existing `raw` format. It defines input/expected file name convention for dataset.